src/Vehicle/RemoteIDManager.cc

```diff
@@ -0,0 +1,424 @@
+        emit commsGoodChanged();
+        qCDebug(RemoteIDManagerLog) << "Receiving ODID_ARM_STATUS from RID device";
+    }
+
+    // Restart the timeout
+    _odidTimeoutTimer.start();
+
+    // CompId and sysId are correct, we can proceed
+    mavlink_open_drone_id_arm_status_t armStatus;
+    mavlink_msg_open_drone_id_arm_status_decode(&message, &armStatus);
+
+    if (armStatus.status == MAV_ODID_GOOD_TO_ARM && !_armStatusGood) {
+        // If good to arm, even if basic ID is not set on GCS, it was set by remoteID parameters, so GCS one would be optional in this case
+        if (!_basicIDGood) {
+            _basicIDGood = true;
+            emit basicIDGoodChanged();
+        }
+        _armStatusGood = true;
+        emit armStatusGoodChanged();
+        qCDebug(RemoteIDManagerLog) << "Arm status GOOD TO ARM.";
+    }
+
+    if (armStatus.status == MAV_ODID_PRE_ARM_FAIL_GENERIC) {
+        _armStatusGood = false;
+        _armStatusError = QString::fromLocal8Bit(armStatus.error);
+        // Check if the error is because of missing basic id
+        if (armStatus.error == QString("missing basic_id message")) {
+            _basicIDGood = false;
+            qCDebug(RemoteIDManagerLog) << "Arm status error, basic_id is not set in RID device nor in GCS!";
+            emit basicIDGoodChanged();
+        }
+        emit armStatusGoodChanged();
+        emit armStatusErrorChanged();
+        qCDebug(RemoteIDManagerLog) << "Arm status error:" << _armStatusError;
+    }
+}
+
+// Function that sends messages periodically
+void RemoteIDManager::_sendMessages()
+{
+    // We only send RemoteID messages if we have it enabled in General settings
+    if (!_settings->enable()->rawValue().toBool()) {
+        return;
+    }
+    
+    // We always try to send System
+    _sendSystem();
+    
+    // only send it if the information is correct and the tickbox in settings is set
+    if (_GCSBasicIDValid && _settings->sendBasicID()->rawValue().toBool()) {
+        _sendBasicID();
+    }
+    
+    // We only send selfID if the pilot wants it or in case of a declared emergency. If an emergency is cleared
+    // we also keep sending the message, to be sure the non emergency state makes it up to the vehicle
+    if (_settings->sendSelfID()->rawValue().toBool() || _emergencyDeclared || _enforceSendingSelfID) {
+        _sendSelfIDMsg();
+    }
+
+    // We only send the OperatorID if the pilot wants it or if the region we have set is europe. 
+    // To be able to send it, it needs to be filled correclty
+    if ((_settings->sendOperatorID()->rawValue().toBool() || (_settings->region()->rawValue().toInt() == Region::EU)) && _operatorIDGood) {
+        _sendOperatorID();
+    }
+
+}
+
+void RemoteIDManager::_sendSelfIDMsg()
+{
+    WeakLinkInterfacePtr weakLink = _vehicle->vehicleLinkManager()->primaryLink();
+
+    if (!weakLink.expired()) {
+        mavlink_message_t       msg;
+        SharedLinkInterfacePtr  sharedLink = weakLink.lock();
```

Comment:
Please use this pattern to avoid race conditions:
```suggestion
    SharedLinkInterfacePtr sharedLink = weakLink.lock();
    if (sharedLink) {
        mavlink_message_t       msg;
```

Reply:
Thanks, it should be fixed now.